core/http-auth/src/main/java/software/amazon/awssdk/http/auth/internal/DefaultAwsV4QueryHttpSigner.java

```diff
@@ -15,27 +15,81 @@
 
 package software.amazon.awssdk.http.auth.internal;
 
+import static software.amazon.awssdk.http.auth.internal.util.CanonicalRequestV2.getCanonicalHeaders;
+import static software.amazon.awssdk.http.auth.internal.util.CanonicalRequestV2.getSignedHeadersString;
+import static software.amazon.awssdk.http.auth.internal.util.CredentialUtils.sanitizeCredentials;
+import static software.amazon.awssdk.http.auth.internal.util.SignerConstant.PRESIGN_URL_MAX_EXPIRATION_DURATION;
+import static software.amazon.awssdk.http.auth.internal.util.SignerUtils.addHostHeader;
+import static software.amazon.awssdk.http.auth.internal.util.SignerUtils.validatedProperty;
+
+import java.time.Duration;
+import java.time.Instant;
+import java.util.List;
 import software.amazon.awssdk.annotations.SdkInternalApi;
+import software.amazon.awssdk.http.SdkHttpRequest;
 import software.amazon.awssdk.http.auth.AwsV4QueryHttpSigner;
-import software.amazon.awssdk.http.auth.spi.AsyncSignRequest;
-import software.amazon.awssdk.http.auth.spi.AsyncSignedRequest;
-import software.amazon.awssdk.http.auth.spi.SyncSignRequest;
-import software.amazon.awssdk.http.auth.spi.SyncSignedRequest;
+import software.amazon.awssdk.http.auth.internal.checksums.ContentChecksum;
+import software.amazon.awssdk.http.auth.internal.util.CanonicalRequestV2;
+import software.amazon.awssdk.http.auth.internal.util.CredentialScope;
+import software.amazon.awssdk.http.auth.internal.util.SignerConstant;
+import software.amazon.awssdk.http.auth.spi.SignRequest;
 import software.amazon.awssdk.identity.spi.AwsCredentialsIdentity;
+import software.amazon.awssdk.identity.spi.AwsSessionCredentialsIdentity;
+import software.amazon.awssdk.utils.Pair;
 
 /**
  * A default implementation of {@link AwsV4QueryHttpSigner}.
  */
 @SdkInternalApi
-public final class DefaultAwsV4QueryHttpSigner implements AwsV4QueryHttpSigner {
+public final class DefaultAwsV4QueryHttpSigner extends DefaultAwsV4HttpSigner implements AwsV4QueryHttpSigner {
 
     @Override
-    public SyncSignedRequest sign(SyncSignRequest<? extends AwsCredentialsIdentity> request) {
-        throw new UnsupportedOperationException();
+    public void addSessionCredentials(SdkHttpRequest.Builder requestBuilder,
+                                      AwsSessionCredentialsIdentity credentials) {
+        requestBuilder.putRawQueryParameter(SignerConstant.X_AMZ_SECURITY_TOKEN, credentials.sessionToken());
+    }
+
+    @Override
+    public void addPrerequisites(SdkHttpRequest.Builder requestBuilder,
+                                 SignRequest<?, ? extends AwsCredentialsIdentity> signRequest,
+                                 ContentChecksum contentChecksum) {
+        Instant requestSigningInstant = validatedProperty(signRequest, SIGNING_CLOCK).instant();
```

Comment:
bug alert: Getting a new instant could mean the date used in scope here could be different than the date used in parent's `sign()` if it crosses date boundary in between. I think the scope should be determined